### PR TITLE
fix: DBTP-989 Make platform-helper generate skip generating AWS Copilot environment pipelines for Terraformed applications

### PR DESCRIPTION
**Pull request by WillGibson**

### CODE REVIEW

File: dbt_platform_helper/utils/validation.py
Comment (WillGibson) [inline dbt_platform_helper/utils/validation.py:240]: I'm not convinced that `environments` or `accounts` are optional any more. Need to confirm that.

@lgarvey @antroy-madetech @ejayesh Do any of you know?

File: dbt_platform_helper/utils/git.py
Comment (WillGibson) [inline dbt_platform_helper/utils/git.py:14]: We will get @benn-dbt's branch merged and merge `main` into here before merging this one.

File: dbt_platform_helper/commands/pipeline.py
Comment (WillGibson) [inline dbt_platform_helper/commands/pipeline.py:53]: Made the error message more helpful because I had to think about it too much.

File: regression_tests/pull_request_tests.sh
Comment (WillGibson) [inline regression_tests/pull_request_tests.sh:32]: We are now guarding slightly more stuff against regressions.

File: dbt_platform_helper/utils/validation.py
Comment (samuele-mattiuzzo) [reply dbt_platform_helper/utils/validation.py:240]: It might have been for different reasons really, but during my recent work on SSO, not having `accounts` wasn't an option

File: dbt_platform_helper/utils/git.py
Comment (samuele-mattiuzzo) [reply dbt_platform_helper/utils/git.py:14]: @WillGibson @benn-dbt will this be the final solution in the end, as opposed to the slightly more readable alternative that was being discussed in Slack?

**Review by samuele-mattiuzzo — APPROVED** [review]

**Review by gabelton — APPROVED** [review]

File: dbt_platform_helper/utils/validation.py
Comment (lgarvey) [reply dbt_platform_helper/utils/validation.py:160]: I haven't done anything with pipelines - so can't comment.

Comment (antroy-madetech) [reply dbt_platform_helper/utils/validation.py:160]: Well, environments *is* optional, but you'll get a pipeline that just builds the tooling and then nothing.
Accounts is mandatory yes.

Comment (antroy-madetech) [reply dbt_platform_helper/utils/validation.py:160]: So environments may as well be mandatory

**Review by antroy-madetech — APPROVED** [review]

File: dbt_platform_helper/utils/validation.py
Comment (WillGibson) [reply dbt_platform_helper/utils/validation.py]: We've made accounts required, but are not sure whether there is still a use case for creating the codebase pipelines without the environment ones existing.

**Review by samuele-mattiuzzo — APPROVED** [review]

**Comment by WillGibson** [comment]
Turns out we need to keep the schema as it was to maintain backwards compatibility wit the 100% AWS Copilot DBT Platform and skip generating the environment pipelines config for Terraform version deploy repositories. So this piece will be getting a rework.

About to dismiss existing approvals.

File: pytest.ini
Comment (samuele-mattiuzzo) [inline pytest.ini:4]:
```suggestion
addopts = --tb=short --numprocesses auto --dist loadgroup
```

File: pytest.ini
Comment (samuele-mattiuzzo) [inline pytest.ini:6]:
```suggestion
;addopts = --tb=short
```

unless, for some reason, this was intentional?

**Review by samuele-mattiuzzo — COMMENTED** [review]
Left a couple of comments / suggestions

File: pytest.ini
Comment (WillGibson) [reply pytest.ini:6]: It's temporary while I rework this subsequent to Friday afternoons end of day conversation. Hold your fire!

File: tests/platform_helper/test_command_pipeline.py
Comment (WillGibson) [inline tests/platform_helper/test_command_pipeline.py:45]: Did a bit of "Arrange - Act - Assert" while I was poking about in this file.

**Review by WillGibson — COMMENTED** [review]
This is ready for review again.

File: dbt_platform_helper/utils/validation.py
Comment (WillGibson) [reply dbt_platform_helper/utils/validation.py:162]: Added to test_load_and_validate_config_valid_file

File: regression_tests/pull_request_tests.sh
Comment (WillGibson) [reply regression_tests/pull_request_tests.sh:32]: It does, and is delegate to by `platform-helper generate`. I have tweaked the verbiage to make it clearer that it's being exercised here.

File: dbt_platform_helper/commands/pipeline.py
Comment (benn-dbt) [reply dbt_platform_helper/commands/pipeline.py:50]: Was there a test that needed to change because the error message changed?

Comment (WillGibson) [reply dbt_platform_helper/commands/pipeline.py:50]: I changed the test first: https://github.com/uktrade/platform-tools/pull/431/files#diff-7a00acf1fa588a66be7746be27e917251534f6a6dd99223d60d7a1736b2ff00eR198

**Review by benn-dbt — APPROVED** [review]
Leaving a discussion about Versioning until we have more time to deal with the fallout. Happy with this PR for now :)

**Pull request merged**
